File: library/src/com/google/maps/android/geojsonkmlabs/Polygon.java
package com.google.maps.android.geojsonkmlabs;

import com.google.android.gms.maps.model.LatLng;

import java.util.ArrayList;
import java.util.List;

public class Polygon implements Geometry {

    private final static String GEOMETRY_TYPE = "Polygon";

    /** {@inheritDoc} */
    public String getGeometryType() {
        return GEOMETRY_TYPE;
    }

    public Object getGeometryObject() { //TODO
        return null;
    }

}

File: android-maps-utils-ktx/src/main/java/com/google/maps/ktx/core/Polygon.kt
package com.google.maps.ktx.core

import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Polygon
import com.google.maps.android.PolyUtil

/**
 * Computes whether or not [latLng] is contained within this Polygon.
 */
fun Polygon.contains(latLng: LatLng): Boolean =
    PolyUtil.containsLocation(latLng, this.points, this.isGeodesic)

/**
 * Checks whether or not [latLng] lies on or is near the edge of this Polygon within a tolerate
 * (in meters) of [tolerance]. The default value
 */
fun Polygon.isOnEdge(latLng: LatLng, tolerance: Double = PolyUtil.DEFAULT_TOLERANCE) =
    PolyUtil.isLocationOnEdge(latLng, this.points, this.isGeodesic, tolerance)
